docs/src/components/home/themes/index.preval.ts

```diff
@@ -0,0 +1,99 @@
+import preval from 'next-plugin-preval';
+import fs from 'fs';
+import { Theme, createTheme } from '@aws-amplify/ui-react';
+import flattenProperties from 'style-dictionary/lib/utils/flattenProperties';
+import { isPlainObject } from 'lodash';
+import terminalTheme from './terminalTheme';
+import classicTheme from './classicTheme';
+import defaultTheme from './defaultTheme';
+import synthwaveTheme from './synthwaveTheme';
+
+const terminalStr = fs.readFileSync(__dirname + '/terminalTheme.ts', {
+  encoding: 'utf-8',
+});
+const classicStr = fs.readFileSync(__dirname + '/classicTheme.ts', {
+  encoding: 'utf-8',
+});
+const defaultStr = fs.readFileSync(__dirname + '/defaultTheme.ts', {
+  encoding: 'utf-8',
+});
+const synthwaveStr = fs.readFileSync(__dirname + '/synthwaveTheme.ts', {
+  encoding: 'utf-8',
+});
+
+const themeCSSVars = (originalTheme, createdTheme, toRet = []) => {
+  for (const name in originalTheme) {
+    if (isPlainObject(originalTheme[name]) && 'value' in originalTheme[name]) {
+      toRet.push(createdTheme[name]);
+    } else if (isPlainObject(originalTheme[name])) {
+      themeCSSVars(originalTheme[name], createdTheme[name], toRet);
+    }
+  }
+  return toRet;
+};
+
+const themeToCSS = (theme: Theme) => {
+  const createdTheme = createTheme(theme);
+
+  const cssVars = themeCSSVars(theme.tokens, createdTheme.tokens);
+  const selector = `:root, [data-amplify-theme]`;
```

Comment:
Is this what we want to scope to?

Reply:
The CSS for the theme switcher is used for Angular and Vue, and that is the default scope our styles.css file uses: https://github.com/aws-amplify/amplify-ui/blob/main/packages/ui/sd.config.ts#L13 

We could probably just go with `:root` since Angular and Vue don't have an element with `[data-amplify-theme]` , but I'm thinking this is more future proof?